### PR TITLE
Forward-port release notes for 0.14.4

### PR DESCRIPTION
**Pull request by jni**
## Description

0.14.4 is out, but the release notes, I think, should be on master, so that they appear on the latest website. (@stefanv correct me if I'm wrong.)

### CODE REVIEW

File: doc/release/release_0.14.rst [hunk 1]
```diff
@@ -1,3 +1,106 @@
...
+- Backport PR #3943 on branch v0.14.x (Update the joblib link in tutorial_parallelization.rst) (#3944)
+
+3 authors added to this release [alphabetical by first name or login]
+---------------------------------------------------------------------
+- Alexandre de Siqueira
+- Juan Nunez-Iglesias
+- Mark Harfouche
+
+
+Announcement: scikit-image 0.14.3
```
Comment (hmaarrfk) [inline doc/release/release_0.14.rst]: opps, we never added these..

**Comment by hmaarrfk** [comment]
I think this should really push us to release 0.16. It is strange that users can use 0.14.4 to overcome such a big bug. If only we had time

**Comment by stefanv** [comment]
How much longer will it take to release 0.16 than 0.14.4?

**Comment by jni** [comment]
I intend to do this in the next couple of days also. ;) @NelleV expressed interest in looking at our release process. =)

**Pull request merged**